### PR TITLE
DS-3937 Add support for styling and rows in the submission-forms.xml

### PR DESCRIPTION
**Pull request by abollini**
This PR
- add row tag in the submissionform-definition so to allow to put multiple fields in the same row in the UI
- add a style tag to provide additional information (as for instance extra css classes) to apply to the field in the UI

it has received an early review here: https://github.com/4Science/DSpace/pull/38

### CODE REVIEW

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SubmissionFormFieldRest.java [hunk 1]
```diff
@@ -85,6 +87,14 @@ public void setHints(String hints) {
         this.hints = hints;
     }
 
+    public String getStyle() {
```
Comment (tdonohue) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SubmissionFormFieldRest.java]: We should add Javadocs to these new methods.  It's not entirely clear what a "Style" is.... is any CSS style valid here?

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SubmissionFormRest.java [hunk 2]
```diff
@@ -53,11 +53,11 @@ public String getCategory() {
         return CATEGORY;
     }
 
-    public List<SubmissionFormFieldRest> getFields() {
-        return fields;
+    public List<SubmissionFormRowRest> getRows() {
```
Comment (tdonohue) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SubmissionFormRest.java]: Again Javadocs should be added to describe what a Row is.

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SubmissionFormRowRest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.app.rest.model;
+
+import java.util.List;
+
+/**
+ * The InputFormRow REST Resource. It is not addressable directly, only used
+ * as inline object in the InputForm resource
```
Comment (tdonohue) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SubmissionFormRowRest.java]: Javadocs could be improved.  This is a `SubmissionFormRow` and not an `InputFormRow`. Javadocs on the public methods below would be nice too.

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/matcher/SubmissionFormFieldMatcher.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath;
+import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasNoJsonPath;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+import org.hamcrest.Matcher;
+
+public class SubmissionFormFieldMatcher {
```
Comment (tdonohue) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/matcher/SubmissionFormFieldMatcher.java]: Javadocs please to describe this class and its public methods

**Review by tdonohue — COMMENTED** [review]
@abollini : Overall this is looking great.  There's just a few JavaDocs missing.

I'll also note that we'll need to update the early submission config documentation at: https://wiki.duraspace.org/display/DSPACE/Configuration+changes+in+the+submission+process   Since this PR is adding a new concept of a `<row>`, we'll need to describe what that is and how it works.

**Review by tdonohue — APPROVED** [review]
👍 This now looks good to me.  The updated documentation notes at https://wiki.duraspace.org/display/DSPACE/Configuration+changes+in+the+submission+process also provide more information on the new "row" concept.  Thanks, @abollini !

**Review by benbosman — COMMENTED** [review]
I couldn’t find time to review this PR yet. If it’s ok, I will review this next week as well to verify everything is resolved

**Comment by tdonohue** [comment]
@benbosman : Will you have a chance to review this soon? I'd like to get this ticket moved along quickly, as I believe there's another (https://github.com/4Science/DSpace/pull/37) from @abollini that is waiting on this one.

**Review by benbosman — COMMENTED** [review]
I had one question on the previous PR (https://github.com/4Science/DSpace/pull/38) which I don't think has been resolved yet.
How can users migrate to the new submission-forms.xml format? I don't see a script to convert their existing submission forms. I may have missed it though.
The remainder looks good for me as well.

**Comment by tdonohue** [comment]
@benbosman : we have a ticket open for that migration work: https://jira.duraspace.org/browse/DS-4046

You are correct that the migration script does not yet exist.  But, I don't think that's a blocker for this PR, as the `submission-forms.xml` file already exists on `master`.

**Review by benbosman — APPROVED** [review]
Thanks @tdonohue for the update
When https://jira.duraspace.org/browse/DS-4046 is completed, this is indeed solved.
I'm approving it as well since there's a plan to solve this

**Comment by tdonohue** [comment]
Merging, as this is at two approvals.  @abollini : I think we're ready for the workflow PR now (https://github.com/4Science/DSpace/pull/38)

**Pull request merged**